test/unit_tests/test_db_upgrade.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+from mock import Mock, patch
+
+from bin import database
+from api import config
+
+CDV = database.CURRENT_DATABASE_VERSION
+
+def test_all_upgrade_scripts_exist():
+    for i in range(1, CDV):
+        script_name = 'upgrade_to_{}'.format(i)
+        assert hasattr(database, script_name)
+
+def test_CDV_was_bumped():
+    script_name = 'upgrade_to_{}'.format(CDV+1)
+    assert hasattr(database, script_name) is False
+
+
+@patch('api.config.get_version', Mock(return_value={'database': 5}))
+def test_get_db_version_from_config():
+    assert database.get_db_version() == 5
+
+
+@pytest.fixture(scope='function')
+def database_mock_setup():
+    setattr(config.db.singletons, 'update_one', Mock())
+    for i in range(1, CDV):
+        script_name = 'upgrade_to_{}'.format(i)
+        setattr(database, script_name, Mock())
+
+@patch('bin.database.get_db_version', Mock(return_value=0))
+def test_all_upgrade_scripts_ran(database_mock_setup):
+    with pytest.raises(SystemExit):
+        database.upgrade_schema()
+    for i in range(1, CDV):
+        script_name = 'upgrade_to_{}'.format(i)
+        assert getattr(database, script_name).called
+
+@patch('bin.database.get_db_version', Mock(return_value=CDV-4))
+def test_necessary_upgrade_scripts_ran(database_mock_setup):
+    with pytest.raises(SystemExit):
```

Comment:
I couldn't find a great way to check which actual sys.exit() code is returned, but ensuring the scripts are called gets us most of the way there.